scripts/utils/setup-roles.ts

```diff
@@ -0,0 +1,145 @@
+    ADMIN_ADDRESS = "0x9b419fd36837558D8A3197a28a5e580AcE44f64F",
+    FEE_CONTROLLER_ADDRESS = "0xE4a1917Ebe8fd2CAFD79799C82aDAa7E81AC6D47",
+    REPAYMENT_CONTROLLER_ADDRESS = "0x23ce21bE3ebd1c86325100460D58d14a1D143E8d",
+    CALL_WHITELIST_ADDRESS = "0x8a12BB999100846B9E56aba4906762353C416952",
+    PUNK_ROUTER_ADDRESS= "0x76391cd8e269f2e8fDcf893E7F5E5781B2Fe2514",
+    ADMIN_ROLE = DEFAULT_ADMIN_ROLE,
+    FEE_CLAIMER_ROLE = DEFAULT_FEE_CLAIMER_ROLE,
+    ORIGINATOR_ROLE = DEFAULT_ORIGINATOR_ROLE,
+    REPAYER_ROLE = DEFAULT_REPAYER_ROLE
+): Promise<void> {
+    const signers: SignerWithAddress[] = await ethers.getSigners();
+    const [admin, adminMultiSig] = signers;
+    const deployer = admin;
+
+    if (!LOAN_CORE_ADDRESS) {
+        throw new Error("Must specify LOAN_CORE_ADDRESS in environment!");
+    }
+
+    if (!admin.address) {
+        throw new Error("Must specify ADMIN_ADDRESS in environment!");
+    }
+
+    if (FEE_CONTROLLER_ADDRESS) {
+        console.log("Fee controller address:", FEE_CONTROLLER_ADDRESS);
+    }
+
+    console.log(SECTION_SEPARATOR);
+
+    const loanCore = await ethers.getContractAt("LoanCore", LOAN_CORE_ADDRESS);
+    const vaultFactory = await ethers.getContractAt("VaultFactory", VAULT_FACTORY_ADDRESS);
+    const lenderNote = await ethers.getContractAt("PromissoryNote", LENDER_NOTE_ADDRESS);
+    const borrowerNote = await ethers.getContractAt("PromissoryNote", BORROWER_NOTE_ADDRESS);
+
+    // grant correct permissions for promissory note
+    // giving to user to call PromissoryNote functions directly
+    for (const note of [borrowerNote, lenderNote]) {
+        await note.connect(admin).initialize(loanCore.address);
+    }
+
+    // grant LoanCore admin fee claimer permissions
+    const updateLoanCoreFeeClaimer = await loanCore.connect(admin).grantRole(FEE_CLAIMER_ROLE, ADMIN_ADDRESS);
+    await updateLoanCoreFeeClaimer.wait();
+
+    console.log(`loanCore has granted fee claimer role: ${FEE_CLAIMER_ROLE} to address: ${ADMIN_ADDRESS}`);
+
+    // grant LoanCore the admin role to enable authorizeUpgrade onlyRole(DEFAULT_ADMIN_ROLE)
+    const updateLoanCoreAdmin = await loanCore.connect(admin).grantRole(ADMIN_ROLE, ADMIN_ADDRESS);
+    await updateLoanCoreAdmin.wait();
+
+    console.log(`loanCore has granted admin role: ${ADMIN_ROLE} to address: ${ADMIN_ADDRESS}`);
+
+    // grant VaultFactory the admin role to enable authorizeUpgrade onlyRole(DEFAULT_ADMIN_ROLE)
+    const updateVaultFactoryAdmin = await vaultFactory.connect(admin).grantRole(ADMIN_ROLE, ADMIN_ADDRESS);
+    await updateVaultFactoryAdmin.wait();
+
+    console.log(`vaultFactory has granted admin role: ${ADMIN_ROLE} to address: ${ADMIN_ADDRESS}`);
+
+    // grant originationContoller the owner role to enable authorizeUpgrade onlyOwner
+    const updateOriginationControllerAdmin = await loanCore.connect(admin).grantRole(ORIGINATOR_ROLE, ORIGINATION_CONTROLLER_ADDRESS);
```

Comment:
It looks like the comment and variable name don't match what is actually happening here - you're granting `OriginationController` the `ORIGINATOR_ROLE` in `LoanCore`. 

We need to do this, but we also need to grant `ADMIN_ADDRESS` the `ADMIN_ROLE` in `OriginationController` (like you do for `VaultFactory`). And it seems like the comment is also mentioning that